api/src/main/java/org/openmrs/module/reporting/report/renderer/ReportTemplateRenderer.java

```diff
@@ -222,17 +222,28 @@ public String getExpressionSuffix(ReportDesign design) {
 	 * @return the value for the report template replacement given the initial value
 	 */
 	public Object getReplacementValue(Object initialValue) {
+
 		Object replacementValue = "";
-		if (initialValue != null) { 
-			if (initialValue instanceof Cohort) {
-				replacementValue = new Integer(((Cohort) initialValue).size());
-			} 
-			else if (initialValue instanceof IndicatorResult) {
-				replacementValue = new Double(((IndicatorResult) initialValue).getValue().doubleValue());
-			}
-			else {
-				replacementValue = initialValue;
+		if (initialValue != null) {
+				if (initialValue instanceof Cohort) {
+					replacementValue = new Integer(((Cohort) initialValue).size());
+				}
+				else if (initialValue instanceof IndicatorResult) {
+				    IndicatorResult ir = (IndicatorResult) initialValue;
+				if(ir.getValue() instanceof Integer){
+					replacementValue = (Integer) ir.getValue();
+				}
+				else if(ir.getValue() instanceof Double) {
+					replacementValue = (Double) ir.getValue();
+				}
+				else {
+					replacementValue = ir.getValue();

```

Comment:
@mseaton this is the else case for when the value is an indicator result but not an integer or a double

Reply:
@mseaton we need this functionality for an upcoiming release, I would love to hear your response to the  changes above thank you